src/LibraryManager/Manifest.cs

```diff
@@ -220,6 +424,20 @@ public void Uninstall(string libraryId, Action<string> deleteFileAction)
             }
         }
 
+        /// <summary>
+        /// Uninstalls the specified library and removes it from the <see cref="Libraries"/> collection.
+        /// </summary>
+        /// <param name="libraryToUninstall">Provider id</param>
+        /// <param name="deleteFileAction"></param>
+        public void Uninstall(ILibraryInstallationState libraryToUninstall, Action<string> deleteFileAction)
```

Comment:
I have also implemented this one...

Reply:
When are you planning to merge your changes to master? 
If your's are merged first, I will rebase and use your implementation.